src/CarbonSDK.ts

```diff
@@ -78,6 +79,7 @@ class CarbonSDK {
 
     this.tmClient = opts.tmClient;
     this.query = new CarbonQueryClient(opts.tmClient);
+    this.insights = new InsightsQueryClient(NetworkConfigs[this.network].insightsUrl);
```

Comment:
can we use `this.networkConfig` instead? so that the user can properly override the config without having to do crazy JS acrobatics.